src/controllers/user-controller.ts

```diff
@@ -1,260 +1,165 @@
+      let email = jwt_decode<Email>(token).email
 
-      const emailTemp = generateEmail(name, 'account', `/?token=${emailToken}`)
+      const existingUser = await User.findOne({ email })
 
-      const data = emailData(email, 'account', emailTemp)
+      if (!existingUser || !existingUser.password) {
+        return res.status(404).json({ message: `User is not registered!` })
+      }
 
-      await sgMail.send(data, false, async (err: any) => {
-        if (err) {
-          return res.status(500).json({
-            message: err.message,
-          })
-        }
+      const hashedPassword = await bcrypt.hash(password, 12)
 
-        const hashedPassword = await bcrypt.hash(password, 12)
-        await User.create({ name, email, password: hashedPassword })
+      await User.updateOne({ email }, { password: hashedPassword })
 
-        return res.status(201).json({
-          message:
-            'User registered successfully! Account verification link sent.',
-        })
-      })
+      return res.status(200).json({ message: 'Password updated successfully' })
     } else {
-      return res.status(401).json({ message: 'Sendgrid api key is missing!' })
+      return res.status(401).json({
+        message:
+          'User is not authorized to change password. Token verification failed.',
+      })
     }
   } catch (error: any) {
     return res.status(500).json({ message: error.message })
   }
 }
 
-export const googleAuth = async (
-  req: RequestBody<RegisterGoogleMemberReq>,
+export const uploadUserImg = async (
+  req: RequestBody<ImageReqBody>,
   res: Response
 ) => {
   try {
-    const { name, email } = req.body
-
-    const existingUser = await User.findOne({ email })
-
-    const token = jwt.sign({ email }, process.env.JWT_SECRET!)
-
-    if (!existingUser) {
-      const newUser = await User.create({ name, email })
-
-      newUser.verified = true
-
-      await newUser.save()
+    const currentUser = await User.findById(req.body.id)
+    if (!currentUser) {
+      return res.status(404).json({ message: 'User not found' })
     }
 
-    return res.status(200).json({
-      token,
-    })
-  } catch (error: any) {
-    return res.status(500).json({ message: error.message })
-  }
-}
-
-export const userAccountActivation = async (
-  req: RequestQuery<EmailActivationReq>,
-  res: Response
-) => {
-  try {
-    const { token } = req.query
-
-    const verified = jwt.verify(token, process.env.JWT_SECRET!)
-
-    if (verified) {
-      let userEmil = jwt_decode<Email>(token).email
-
-      const existingUser = await User.findOne({ email: userEmil })
-
-      if (!existingUser) {
-        return res.status(404).json({ message: `User is not registered yet!` })
-      }
+    if (req.body.fileValidationError) {
+      return res.status(422).json({ message: 'Upload only image' })
+    }
 
-      await User.updateOne({ email: userEmil }, { verified: true })
+    if (req.file) {
+      currentUser.image = req.file.path.substring(7)
```

Comment:
what does this 7 means?

Reply:
when the server gets a request with an image file, `req.file.path` is something like that - 'public/images/users/image.png'. I decided to save image URI-s on MongoDB with different standards - 'images/users/image.png'. What `req.file.path.substring(7)` does is that it removes the first 7 characters ('public/') from the 'public/images/users/image.png' in order to get 'images/users/image.png' and save it on the server.